File: frontend/Artform/types/review.ts
// types/review.ts

export type Review = {
  reviewId: number;
  modelId: number;
  modelName: string;
  reviewImageName: string;
  presignedUrl: string;
  userId: number;
  userName: string;
  content: string;
  createdAt: string;
};

export type Reviews = {
  data: Review[];
  reviewCount: number;
};

File: backend/artform-core/src/main/java/com/d103/artformcore/entity/Review.java
package com.d103.artformcore.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "review")
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long reviewId;

    @ManyToOne
    @JoinColumn(name = "model_id", nullable = false)
    private Model model;

    @Column(nullable = true)
    private String reviewImageName;

    @Column(nullable = false)
    private Long userId;

    @Column(length = 255)
    private String content;

    @Column(nullable = false)
    @CreationTimestamp
    private LocalDateTime createdAt = LocalDateTime.now();
}
